sdk/cosmosdb/cosmos/src/request/ResourceResponse.ts

```diff
@@ -11,6 +17,15 @@ export class ResourceResponse<TResource> {
     public readonly statusCode: StatusCode,
     public readonly substatus?: SubStatusCode
   ) {}
+  public get cosmosDiagnosticsRegionsContacted(): DiagnosticHeaders {
+    return {};
+  }
+  public get cosmosDiagnostisDurationInMs(): number {
+    return getdiagnosticsdurationMilliseconds();
+  }
+  public get cosmosDiagnostics(): string {
+    return getCosmosDiagnosticsToString();
```

Comment:
If my understanding is correct - getCosmosDiagnosticsToString will reset the diagnosticHeaders[] - so the getxxx has teh side-effect to aslo reset diagnostics and make it impossible to retrieve it again. This seems counter-intuitive and should be changed.

Reply:
Never mind - I misread the getCosmosDiagnosticsToString method initially.

Reply:
:)